patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch

```diff
@@ -57,153 +57,169 @@
+ #include "chrome/grit/generated_resources.h"
+ #include "components/enterprise/common/proto/connectors.pb.h"
+ #include "components/policy/core/browser/url_util.h"
+ #include "components/policy/core/common/chrome_schema.h"
+ #include "components/prefs/pref_service.h"
+-#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
+ #include "components/safe_browsing/core/features.h"
+ #include "components/url_matcher/url_matcher.h"
+ #include "content/public/browser/web_contents.h"
 --- a/chrome/browser/extensions/api/safe_browsing_private/safe_browsing_private_event_router.cc
 +++ b/chrome/browser/extensions/api/safe_browsing_private/safe_browsing_private_event_router.cc
-@@ -262,6 +262,7 @@ void SafeBrowsingPrivateEventRouter::OnD
+@@ -273,6 +273,7 @@ void SafeBrowsingPrivateEventRouter::OnD
      event_router_->BroadcastEvent(std::move(extension_event));
    }
  
 +#if defined(FULL_SAFE_BROWSING)
    if (!IsRealtimeReportingEnabled())
      return;
  
-@@ -293,6 +294,7 @@ void SafeBrowsingPrivateEventRouter::OnD
+@@ -304,6 +305,7 @@ void SafeBrowsingPrivateEventRouter::OnD
            },
            params.url, params.file_name, params.download_digest_sha256,
            params.user_name, mime_type, content_size));
 +#endif // FULL_SAFE_BROWSING
  }
  
  void SafeBrowsingPrivateEventRouter::OnSecurityInterstitialShown(
-@@ -320,6 +322,7 @@ void SafeBrowsingPrivateEventRouter::OnS
+@@ -331,6 +333,7 @@ void SafeBrowsingPrivateEventRouter::OnS
      event_router_->BroadcastEvent(std::move(extension_event));
    }
  
 +#if defined(FULL_SAFE_BROWSING)
    if (!IsRealtimeReportingEnabled())
      return;
  
-@@ -349,6 +352,7 @@ void SafeBrowsingPrivateEventRouter::OnS
+@@ -360,6 +363,7 @@ void SafeBrowsingPrivateEventRouter::OnS
            },
            params.url, params.reason, net_error_code, params.user_name,
            event_result));
 +#endif // FULL_SAFE_BROWSING
  }
  
  void SafeBrowsingPrivateEventRouter::OnSecurityInterstitialProceeded(
-@@ -376,6 +380,7 @@ void SafeBrowsingPrivateEventRouter::OnS
+@@ -387,6 +391,7 @@ void SafeBrowsingPrivateEventRouter::OnS
      event_router_->BroadcastEvent(std::move(extension_event));
    }
  
 +#if defined(FULL_SAFE_BROWSING)
    if (!IsRealtimeReportingEnabled())
      return;
  
-@@ -397,6 +402,7 @@ void SafeBrowsingPrivateEventRouter::OnS
+@@ -408,6 +413,7 @@ void SafeBrowsingPrivateEventRouter::OnS
              return event;
            },
            params.url, params.reason, net_error_code, params.user_name));
 +#endif // FULL_SAFE_BROWSING
  }
  
  void SafeBrowsingPrivateEventRouter::OnAnalysisConnectorResult(
-@@ -433,6 +439,7 @@ void SafeBrowsingPrivateEventRouter::OnD
-     const std::string& trigger,
+@@ -420,6 +426,7 @@ void SafeBrowsingPrivateEventRouter::OnA
+     const enterprise_connectors::ContentAnalysisResponse::Result& result,
      const int64_t content_size,
      safe_browsing::EventResult event_result) {
 +#if defined(FULL_SAFE_BROWSING)
```

Comment:
This `#if defined(FULL_SAFE_BROWSING)` got moved from the start of `SafeBrowsingPrivateEventRouter::OnDangerousDeepScanningResult()` to `SafeBrowsingPrivateEventRouter::OnAnalysisConnectorResult()`. Since the corresponding `#endif` is at the end of `SafeBrowsingPrivateEventRouter::OnDangerousDeepScanningResult()`, the net effect of this is that there is no implementation of `SafeBrowsingPrivateEventRouter::OnDangerousDeepScanningResult()`. Since this is ungoogled-chromium, this function is not likely to be called, so this might not cause issues immediately, but it might cause linker errors down the road. I would recommend correctly scoping the `#if defined ... #endif` block; note that if you do this, you will likely just need to add two lines: an `#endif` at the end of `OnAnalysisConnectorResult()` and an `#if defined(FULL_SAFE_BROWSING)` at the start of `OnDangerousDeepScanningResult()`.

Reply:
@tangalbert919 Please confirm if this change is needed. Otherwise this LGTM.